distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFtp.java

```diff
@@ -95,61 +113,179 @@ public void afterExam() throws Exception {
         }
     }
 
-    @After
-    public void tearDown() throws Exception {
-        disconnect();
-        client = null;
+    /**
+     * Simple test verifying FTP client can be connected and successfully complete SSL handshake
+     * with FTP server when clientAuth = "want"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpWantClientAuth() throws Exception {
+        setClientAuthConfiguration(CLIENT_AUTH_WANT);
+        FTPClient client = createInsecureClient();
+        assertTrue(client.sendNoOp());
+        disconnectClient(client);
     }
 
     /**
-     * Simple test verifying FTP client can be connected and configured successfully
-     * 
+     * Simple test verifying FTPS client with keystore can be connected and successfully complete
+     * SSL handshake with FTP server when clientAuth = "want"
+     *
      * @throws Exception
      */
     @Test
-    public void testFtp() throws Exception {
-        client = createInsecureClient();
+    public void testFtpsWithKeystoreWantClientAuth() throws Exception {
+        setClientAuthConfiguration(CLIENT_AUTH_WANT);
+        FTPSClient client = createSecureClient(true);
         assertTrue(client.sendNoOp());
+        disconnectClient(client);
+    }
+
+    /**
+     * Simple test verifying FTPS client with keystore can be connected and successfully complete
+     * SSL handshake with FTP server when clientAuth = "need"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsWithKeystoreNeedClientAuth() throws Exception {
+        setClientAuthConfiguration(CLIENT_AUTH_NEED);
+        FTPSClient client = createSecureClient(true);
+        assertTrue(client.sendNoOp());
+        disconnectClient(client);
+    }
+
+    /**
+     * Simple test verifying FTPS client without keystore can be connected and successfully complete
+     * SSL handshake with FTP server when clientAuth = "want"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsWithoutKeystoreWantClientAuth() throws Exception {
+        setClientAuthConfiguration(CLIENT_AUTH_WANT);
+        FTPSClient client = createSecureClient(false);
+        assertTrue(client.sendNoOp());
+        disconnectClient(client);
+    }
+
+    /**
+     * Simple test verifying FTPS client without keystore can be connected and successfully complete SSL handshake
+     * with FTP server when clientAuth = "need"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsWithoutKeystoreNeedClientAuth() throws Exception {
+        setClientAuthConfiguration(CLIENT_AUTH_NEED);
+        FTPSClient client = null;
+        try {
+            client = createSecureClient(false);
+            disconnectClient(client);
+            fail("Expected fail to complete SSL handshake between FTPS client and FTP server");
+        } catch (SSLException e) {
+            disconnectClient(client);
```

Comment:
Do you need to attempt to disconnect the client if he was never able to complete the handshake? If not, I think you can remove this try catch block and replace it with `@Test(expected= SSLException.class)`